Network Class/Network.cpp

```diff
@@ -178,6 +176,60 @@ bool backProp(int index)
 	return correct;
 }
 
+// checks if Network output matches expected
+// @param - takes a matrix of expected vals (could be changed to vector though or process both if need be)
+// returns <0 if correct, >0 if incorrect and a 0 if ambiguous
+bool Network::compareOutput(const Matrix& expectedValues) 
+{
+	int lastInd = numLayers - 1;
+	int lastSize = layerSizes[lastInd];
+
+	// indices at which there is the biggest activation or a 1 for the netowrk or expected vals respectively
+	int biggestI = -1; int expectedI = -1;
+
+	// biggest starts at 0 (not MIN) because any number < 0 might as well be 0 for these purposes
+	// numBiggest is number of times we encounter the biggest element in output layer
+	double biggest = 0;	int numBiggest = 0;
+
+	// out of place random check to make sure it's being fed good data
+	if (expectedValues.size() != lastSize)
+		return 0;
+	for (int i = 0; i < lastSize; i++)
+	{
+		double output = activations[lastInd](i, 0);
+		if (output > biggest)
+		{
+			biggestI = i;
+			biggest = output;
+			numBiggest = 0; // we found a new biggest so numBiggest's previous data is invalid
+		}
+		else if (output == biggest) 
+			numBiggest++;
+
+		if (expectedValues(i, 0) == 1 && expectedI == -1)
+		{
+			if (expectedI == -1) // if it's not been set before..
+				expectedI = i;
+			else				 // it it has it's bad data
+				expectedI = -2;
+		}
+	}
+
+	// the return statements here can and should change depending on what we agree is good, bad, or ambiguous data
+	if (biggestI == -1 || expectedI <= -1) // everything was smaller than 0 or expected is bad data (probably should throw excpetion instead actually)
+		return 0;
+	 
+	// now we can be sure biggestI, expectedI >=0
+
+	if (numBiggest > 0) // if there's more than 1 biggest, ambiguous data. don't even care if the indices match
```

Comment:
all the "bad data" cases return the same value. It may be worthwhile to distinguish the errors/ambiguities for troubleshooting.